lua/autorun/sh_cfc_spawnpoint.lua

```diff
@@ -0,0 +1,120 @@
+CFC_SpawnPoints = CFC_SpawnPoints or {}
+
+if cleanup then
+    cleanup.Register( "sent_spawnpoint" )
+end
+
+CreateConVar( "sbox_maxsent_spawnpoint", 1, { FCVAR_ARCHIVE, FCVAR_REPLICATED }, "The max number of spawn points per player.", 0, 100 )
+
+local REMOVAL_WINDOW = CreateConVar( "cfc_spawnpoints_removal_window", 30, { FCVAR_ARCHIVE, FCVAR_REPLICATED }, "Player/point cooldowns only apply if a previous spawn point was removed in the past X seconds. 0 to not alter cooldowns.", 0, 1000 )
+
+
+----- GLOBAL FUNCTIONS -----
+
+--[[
+    - Determines whether or not a player is considered 'friendly' to a spawn point.
+        - i.e. they can link to it, if no cooldowns or other restrictions block them.
+    - Returns friendly, failReason
+    - You can override this function in InitPostEntity if you need a different 'friendliness' check.
+--]]
+function CFC_SpawnPoints.IsFriendly( spawnPoint, ply )
+    if not CPPI then
+        if spawnPoint:GetCreatingPlayer() == ply then return true end
+
+        return false, "You can only link to your own Spawn Points."
+    end
+
+    local owner = spawnPoint:CPPIGetOwner()
+    if ply == owner then return true end
+
+    local friends = owner.CPPIGetFriends and owner:CPPIGetFriends()
+
+    if not friends or friends == CPPI.CPPI_DEFER or friends == CPPI.CPPI_NOTIMPLEMENTED then
+        return false, "You can only link to your own Spawn Points."
+    end
+
+    if table.HasValue( friends, ply ) then return true end
+
+    return false, "You are not buddied with the Spawn Point's owner."
+end
```

Comment:
Would probably be good to have a hook in here somewhere too (squad addons or something)

Reply:
In the readme, it's noted that this function is meant to be overridden as needed, and cfc_pvp already does this. I personally think spawnpoint friendliness needs too specific of a check to make sense with hooks, but I can change it if you have a good idea on how to handle it.

Reply:
:+1: